p2p/src/authenticated/network.rs

```diff
@@ -18,19 +18,14 @@ use rand::{CryptoRng, Rng};
 use std::marker::PhantomData;
 use tracing::{debug, info, warn};
 
-// The maximum overhead of encoding a `message: Bytes` into a protobuf `message Message`
-// Should be at most 18 bytes for messages under 4GB, but we add a bit of padding.
-//
-// The byte overhead is calculated as follows:
-// - 1  Data field number
-// - 5* Data length varint
-// - 1  Channel field number
-// - 5  Channel value varint
-// - 1  Message field number
-// - 5* Message length varint
-//
-// (*) assumes that the length is no more than 4GB
-const PROTOBUF_OVERHEAD: usize = 64;
+/// The maximum overhead of encoding a `message: Bytes` into a full
+/// [Payload](super::types::Payload), in bytes.
+///
+/// The byte overhead is calculated as the sum of the following:
+/// - 1: Payload enum value
+/// - 5: Channel varint
+/// - 5: Message length varint (assumes length is no more than 4GB)
+const MAX_PAYLOAD_OVERHEAD: usize = 11;
```

Comment:
Can we pull this from the `types` or `wire` file instead of maintaining it here?

I think that would also provide a great opportunity to assert that an example messages has less than this overhead over the raw payload.